### DISCUSSION TITLE
Signature template for all users

### DISCUSSION THREAD

**Discussion by cm-schl** [the-djmaze/snappymail#771] [Ideas]
I would like to start a discussion on a topic that would be great for companies with many users. When SnappyMail is used in this environments a centralized configuration for signatures is often needed to meet Corporate Identity requirements (same design, font etc.).

I've made some thoughts on this and think that an additional plugin could solve this needs..?

Points that I have identified as needed:

- Configuration dialog where the default signature with placeholders for dynamic contents (see below) can be defined (with HTML editor like in the signature settings of every single user). At the moment a plugin can not have an editor inside it's config dialog (I think)
- Possibility to activate this template for every user. My use case is that every user can have the same signature template - probably other companies need the possibility to define signature for user groups?
- Fetching dynamic parts of the signature (name, surname, department, role, phone number etc) and substituting the placeholders with the retrieved data. I think in many situations an LDAP query would be useful.

As so often at the beginning it seams that this all can be done within some hours but I think there could arise different problems I actually have not identified 🙂. It seams that also others are considering to use SnappyMail in a professional environment (@the-djmaze, @Mer0me) so maybe we could collect some thoughts on this topic?

**Comment by the-djmaze** [comment]
There's an open issue regarding templates #152
Basically it is for responses and not for signatures.

Although there is a `filter.account` hook, there is none for the account identities.
What if there is a new hook for identities?
A plugin can then:
1. detect account domain
2. on match, inject identity

**Reply by cm-schl** [reply]
I definitely have not thought about this fact when the injection has to be done...
Would this eventually new hook also mean that after login a user could change from one account to an other and the data needed to replace the placeholders could be retrieved on changing the account?

The additional point to check the domain is a good idea - this would extend the second point on my list.
